File: Java/dependency-injection/_04_bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/Inbound/UdpSocket.java
package com.github.grzesiek_galezowski.BastardInjection.Inbound;

class UdpSocket implements InputSocket {
  public boolean receive(byte[] frameData) {
    //todo generate random the java way new Random().NextBytes(frameData);
    return true;
  }

  public void close() {
    System.out.println("Disposed");
  }
}

File: CSharp/DependencyInjection/DependencyInjection/DependencyInjectionAfter/_1_ControlFreakRefactoredToDependencyInjection/Inbound/UdpSocket.cs
using System;

namespace DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Inbound
{
  internal interface IInboundSocket
  {
    bool Receive(out byte[] frameData);
  }

  class UdpSocket : IInboundSocket
  {
    public bool Receive(out byte[] frameData)
    {
      frameData = new byte[100];
      new Random().NextBytes(frameData);
      return true;
    }
  }
}